validator/src/journal/candidate_block.rs

```diff
@@ -39,6 +39,27 @@ use pylogger;
 
 use scheduler::Scheduler;
 
+#[derive(PartialEq, Debug)]
+pub enum BatchInjectionStage {
+    BlockStart,
+    BeforeBatch,
+    AfterBatch,
+    BlockEnd,
+    BlockUnderFormation,
```

Comment:
Could you please add a comment on how is this used?
Is this a default not-set value?